src/components/Transactions/TransactionRow.tsx

```diff
@@ -37,8 +38,21 @@ export const TransactionRow = ({ tx, compact }: { tx: Tx; compact?: boolean }) =
   const [isOpen, setIsOpen] = useState(false)
   const toggleOpen = () => setIsOpen(!isOpen)
   const sentTx = tx.type === chainAdapters.TxType.Send
+  const receivedTx = tx.type === chainAdapters.TxType.Receive
+  const tradeTx = tx.type === chainAdapters.TxType.Trade
   const symbol = tx?.chainSpecific?.token?.symbol ?? asset?.symbol
 
+  const allAssets = useSelector((state: ReduxState) => state.assets)
+
+  // TODO compare using caip ids
+  // Cant do this yet because unchained doesnt only returns symbol with trade data
+  const buyAsset = Object.values(allAssets).filter(
```

Comment:
Might as well use `.find()` so we can just find first (and typically only) match and bail

Reply:
And maybe don' even do the search if there is no trade data to save some compute

Reply:
I changed it to find but the other is so insignificant to save compute i didnt bother